logger.py

```diff
@@ -112,9 +113,14 @@ def setup_logger(name, level=logging.INFO, logging_file="latest.log"):
     ch.setFormatter(StreamLogFormatter())
     logger.addHandler(ch)
 
-    if logging_file:
-    # Create file handler
-        fh = logging.FileHandler(logging_file, mode="w")
+    if logging_settings.get("file", False):
+        # Create file handler
+        fh = handlers.TimedRotatingFileHandler(
+            logging_settings.get("file", "latest.log"),
+            interval=logging_settings.get("rollover_hours", 24),
+            when="H",
```

Comment:
I was imagining doing `midnight` for the `when` so that the log would rollover at midnight. As a result, we could remove the `rollover_hours` variable.

Reply:
Changed this now to midnight. I was not sure which one you'd rather wanted at first.